src/circuit_writer/writer.rs

```diff
@@ -175,6 +174,47 @@ impl<B: Backend> CircuitWriter<B> {
                     fn_env.pop();
                 }
             }
+            StmtKind::IteratorLoop {
+                var,
+                iterator,
+                body,
+            } => {
+                let iterator_expr = self
```

Comment:
maybe it is better to call it as `iterator_var`, since it is of `VarOrRef` after the expression has been computed.